account_invoice_constraint_chronology_in_date_range/models/account_move.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2024 Foodles (https://www.foodles.co/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+from odoo.osv import expression
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _get_older_conflicting_invoices_domain(self):
+        self.ensure_one()
+        if not self.invoice_date or not self.journal_id.sequence_id.use_date_range:
+            return super()._get_older_conflicting_invoices_domain()
+        date_range_id = self.journal_id.sequence_id._get_current_sequence(
```

Comment:
non blocking point, it's personal style preference, when I store a record in a vairable I avoid to suffix by `_id` I don't think it's an OCA conventions either !

```suggestion
        date_range = self.journal_id.sequence_id._get_current_sequence(
```